apps/timetable/admin/ui/js/index.js

```diff
@@ -0,0 +1,411 @@
+            'data': null
+        }, $('#gh-modules-container'));
+        $('.gh-tripos-help').show();
+    };
+
+
+    /////////////////
+    //  UTILITIES  //
+    /////////////////
+
+    /**
+     * Log in using the local authentication strategy
+     *
+     * @param  {Event}    ev    The jQuery event
+     * @private
+     */
+    var doLogin = function(ev) {
+
+        // Prevent the form from being submitted
+        ev.preventDefault();
+
+        // Collect and submit the form data
+        var formValues = _.object(_.map($(this).serializeArray(), _.values));
+        gh.api.authenticationAPI.login(formValues.username, formValues.password, function(err) {
+            if (!err) {
+                window.location = '/admin';
+            } else {
+                gh.api.utilAPI.notification('Login failed', 'Logging in to the application failed', 'error');
+            }
+        });
+    };
+
+    /**
+     * Log the current user out
+     *
+     * @param  {Event}    ev    The jQuery event
+     * @private
+     */
+    var doLogout = function(ev) {
+
+        // Prevent the form from being submitted
+        ev.preventDefault();
+
+        gh.api.authenticationAPI.logout(function(err) {
+            if (!err) {
+                window.location = '/admin';
+            } else {
+                gh.api.utilAPI.notification('Logout failed', 'Logging out of the application failed', 'error');
+            }
+        });
+    };
+
+
+    /////////////
+    //  VIDEO  //
+    /////////////
+
+    /**
+     * Initialise the video
+     *
+     * @private
+     */
+    var initVideo = function() {
+        if (gh.api.utilAPI.localDataStorage().get('hideVideo')) {
+            return hideVideo();
+        }
+        showVideo();
+
+        // Do not show the video the next time
+        gh.api.utilAPI.localDataStorage().store('hideVideo', true);
+    };
+
+    /**
+     * Hide the help video
+     *
+     * @private
+     */
+    var hideVideo = function() {
+        $('.gh-video').hide();
+        $('.gh-show-video').show();
+        return false;
+    };
+
+    /**
+     * Show the help video
+     *
+     * @private
+     */
+    var showVideo = function() {
+        $('.gh-video').show();
+        $('.gh-show-video').hide();
+        return false;
+    };
+
+    /**
+     * Play the help video
+     *
+     * @private
+     */
+    var playVideo = function() {
+        return false;
+    };
+
+
+    //////////////////////
+    //  INITIALISATION  //
+    //////////////////////
+
+    /**
+     * Add bindings to various elements on the page
+     *
+     * @private
+     */
+    var addBinding = function() {
+
+        // Video events
+        $('body').on('click', '.gh-show-video', showVideo);
+        $('body').on('click', '.gh-hide-video', hideVideo);
+        $('body').on('click', '.gh-play-video', playVideo);
+
+        // Login and logout
+        $('body').on('submit', '.gh-signin-form', doLogin);
```

Comment:
This is probably something I did originally but why is `gh-signin-form` a class instead of an ID?